scout/server/utils.py

```diff
@@ -106,6 +106,49 @@ def institute_and_case(store, institute_id, case_name=None):
     return institute_obj
 
 
+def rank_score(variant, rank_model):
+    """Calculate rank ranges for the variant and its corresponding rank model."""
+
+    def get_scores(field):
+        return [
+            int(v.get("score"))
+            for v in field.values()
+            if isinstance(v, dict) and isinstance(int(v.get("score")), (int))
+        ]
+
+    def get_rank(model, info_fields, variant, category):
+        fields = [f for f in info_fields if f.get("category") == category]
+        scores = [get_scores(field) for field in fields]
+        flattend_scores = [n for v in scores for n in v]
+        # finds the already pre-calculated score
+        rank_score = next(
+            (r for r in variant.get("rank_score_results") if r.get("category") == category), None
```

Comment:
This fails if no result is available in the category. next(None) does not work. If you really, really want to use the list comprehension anyway, perhaps cast it in a containing list(). Otw expand to a loop with some conditionals to handle non existing values?

Reply:
good catch! but I'm a bit confused on the explanation.

you mean it crashes 💥 on the later rank_score.get? or crashes when `category` is None? or it fails if the variant does not have a score? or something else? :D

Reply:
Probably me too, but either I guess. If the inner expression in the next() evaluates to none, you should get like a "NoneType is not an iterable", right? I'm no longer at the computer so won't check how likely either of the inner ones is to be missing - or if they are well checked for existence beforehand, but at the very least you saw fit to add a fallback None, so I'm guessing it can happen. 😅